File: src/main/java/com/dansoftware/libraryapp/gui/info/dependency/DependenciesWindow.java
package com.dansoftware.libraryapp.gui.info.dependency;

import com.dansoftware.libraryapp.gui.util.LibraryAppStage;
import javafx.stage.Modality;
import javafx.stage.Window;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class DependenciesWindow extends LibraryAppStage {

    public DependenciesWindow(@NotNull DependencyTable dependencyTable, @Nullable Window owner) {
        super("window.dependencies.title", dependencyTable);
        initOwner(owner);
        initModality(Modality.APPLICATION_MODAL);
    }
}

File: src/main/java/com/dansoftware/libraryapp/gui/info/dependency/DependenciesWindow.kt
package com.dansoftware.libraryapp.gui.info.dependency

import com.dansoftware.libraryapp.gui.window.BaseWindow
import javafx.stage.Modality
import javafx.stage.Window

/**
 * A [Window] that is used for showing a [DependencyTable].
 *
 * @author Daniel Gyorffy
 */
class DependenciesWindow(dependencyTable: DependencyTable, owner: Window?) :
    BaseWindow<DependencyTable>("window.dependencies.title", dependencyTable) {
    init {
        initOwner(owner)
        initModality(Modality.APPLICATION_MODAL)
        isResizable = false
    }
}